### ISSUE TITLE
faraday caching?

### ISSUE THREAD

**Issue by MichaelSid** [jaimeiniesta/metainspector#143]
Hi,

I see that there are some faraday options included, but I can't find a way to add Faraday http caching to the MetaInspector call. 

I need this to reduce the response time for subsequent calls on the same url. 

Thanks,
Michael

**Comment by jaimeiniesta** [comment]
Hi,

We do the http request on initialize, that is:

```ruby
page = MetaInspector.new('https://github.com') # here we do the request

# From now on, the http request is memoized on that instance,
# so no more requests will be done:

page.title
page.description
page.keywords

# etc...
```

That being said, if you want to add support for cache that would persist between different initializations, I'll welcome a pull request :)

It looks like the [Faraday Http Cache](https://github.com/plataformatec/faraday-http-cache) could be easily integrated, as we're already passing [faraday options](https://github.com/jaimeiniesta/metainspector/blob/master/lib/meta_inspector/request.rb#L20) to the request. We could check if those options include a `:store`, and in that case, tell Faraday to use it.

**Comment by jaimeiniesta** [comment]
@MichaelSid can you take a look at this PR? https://github.com/jaimeiniesta/metainspector/pull/146

This PR implements the Faraday HTTP Cache, but we're not sure if this feature is worth adding, as it just caches the page content, but not the request, that is, the DNS and TCP request phase is still performed.

**Comment by MichaelSid** [comment]
Hi @jaimeiniesta @jschwindt 

Great I'll check it out ASAP and get back to you, thanks. Looking at the PR, it looks very similar to the PR I was working on - I guess I was beaten to it :)

Anyway correct me if I'm wrong but, isn't it a good thing that it still performs the request again, so it can check if the contents have been modified? So there's no risk of repeatedly serving an outdated page?

**Comment by jaimeiniesta** [comment]
@MichaelSid I think it depends on the use case, you might want to have a complete cache (including the request) so that, say, requests to the same URL within 1 minute are handled by the cache.

**Comment by jaimeiniesta** [comment]
Faraday::HttpCache support released in 4.7.0, thanks!
